src/searchBar.js

```diff
@@ -0,0 +1,21 @@
+import { displayOneCategory } from "./displayUI";
+import { displayAllCategories } from "./displayUI";
+import { singletonInstance } from './mainClass';
+
+singletonInstance;
+//Search tab functionality
+export function searchCategory() {
+  const searchBar = document.querySelector("#searchBar");
+
+  displayAllCategories(singletonInstance);
+
+  searchBar.addEventListener("input", (e) => {
+    e.preventDefault();
+    document.querySelector("#categories").innerHTML = "";
+    singletonInstance.categories.forEach((el) => {
```

Comment:
Search logic should be part of model class, that could return result for UI here. Or even search to be property of class, and to we have method getFilteredData which is used by UI and applies filters